tests/task/test_demo_task_creation.py

```diff
@@ -0,0 +1,158 @@
+      return levels.sum() >= number
+
+    def equipment(scenario: Scenario):
+      scenario.add_tasks(EquipmentLevel(number=1 )*Tier.EASY, groups='agents')
+      scenario.add_tasks(EquipmentLevel(number=5 )*Tier.NORMAL, groups='agents')
+      scenario.add_tasks(EquipmentLevel(number=10)*Tier.HARD, groups='agents')
+      return scenario.tasks
+    
+    @t.define_predicate
+    def CombatSkill(gs, subject, lvl):
+        return t.OR(p.AttainSkill(subject, skill.Melee, lvl, 1),
+                  p.AttainSkill(subject, skill.Range, lvl, 1),
+                  p.AttainSkill(subject, skill.Mage, lvl, 1))
+
+    def combat(scenario: Scenario):
+      scenario.add_tasks(CombatSkill(lvl=2)*Tier.EASY, groups='agents')
+      scenario.add_tasks(CombatSkill(lvl=3)*Tier.NORMAL, groups='agents')
+      scenario.add_tasks(CombatSkill(lvl=4)*Tier.HARD, groups='agents')
+      return scenario.tasks
+
+    @t.define_predicate
+    def ForageSkill(gs, subject, lvl):
+        return t.OR(p.AttainSkill(subject, skill.Fishing, lvl, 1),
+                  p.AttainSkill(subject, skill.Herbalism, lvl, 1),
+                  p.AttainSkill(subject, skill.Prospecting, lvl, 1),
+                  p.AttainSkill(subject, skill.Carving, lvl, 1),
+                  p.AttainSkill(subject, skill.Alchemy, lvl, 1))
+  
+    def foraging(scenario: Scenario):
+      scenario.add_tasks(ForageSkill(lvl=2)*Tier.EASY)
+      scenario.add_tasks(ForageSkill(lvl=3)*Tier.NORMAL)
+      scenario.add_tasks(ForageSkill(lvl=4)*Tier.HARD)
+      return scenario.tasks
+
+    # Demonstrate task scenario definition API
+    def all_tasks(scenario: Scenario):
+      player_kills(scenario)
+      exploration(scenario)
+      equipment(scenario)
+      combat(scenario)
+      foraging(scenario)
+      return scenario.tasks
+
+    # Test rollout
+    task_generators = [player_kills, exploration, equipment, combat, foraging, all_tasks]
+    for tg in task_generators:
+      config = ScriptedAgentTestConfig()
+      env = Env(config)
+      scenario = Scenario(config)
+      tasks = tg(scenario)
+      env.change_task(tasks)
+      for _ in range(30):
+        env.step({})
+
+    # DONE
+  
+  def test_player_kill_reward(self):
+    """ Reward 0.1 per player defeated, 1 for first and 3rd kills
+    """
+    config = ScriptedAgentTestConfig()
+    env = Env(config)
+    scenario = Scenario(config)
+
+    # PARTICIPANT WRITES
+    # ====================================
+    @t.define_task
+    def KillTask(gs: GameState,
+                 subject: Group):
+      score = len(getattr(subject.event, 'PLAYER_KILL')) * 0.1
+      if p.CountEvent(subject, 'PLAYER_KILL', 1)(gs) == 1.0:
+        score += 0.9
+      if p.CountEvent(subject, 'PLAYER_KILL', 3)(gs) == 1.0:
+        score += 1
+      return score
+
+    scenario.add_tasks(lambda agent: KillTask(agent), groups='agents')
+    # ====================================
+
+    # Test Reward
+    env.change_task(scenario.tasks)
```

Comment:
I assume env.change_task() is also doing reset. @jsuarez5341 @RyanNavillus is this a common way of doing this?